### ISSUE TITLE
(URGENT) Update to IEEE SA bibliographic item stages

### ISSUE THREAD

**Issue by opoudjis** [relaton/relaton-ieee#26]
From https://standards.ieee.org/develop/develop-standards/process/ , we know the following statuses apply for IEEE:

Approved Project – An initial project request is approved, in stages of group formation.
Active Project – An active standards development project.
Withdrawn Project – A cancelled standards development project.
Approved Standard – The standard is approved and published for public use.
Withdrawn Standard – The standard is no longer market relevant or active.
Superseded Standard – The standard has been replaced by a new standard.

These fall into three overall status groups, given in https://standards.ieee.org/about/policies/opman/sect1/:

Developing = Draft: approved project, active project, withdrawn project
Active: approved standard
Inactive: withdrawn standard, superseded standard

The statuses being gathered by screenscraping are combining the two lists: we are getting Inactive-Withdrawn

We do need a consistent vocabulary of statuses. Could you please review the statuses you are gathering by screen scraping, and come up with a consistent list, that we can also mark up Metanorma documents with? I can accept inactive-withdrawn, but I need to know that inactive- is being used consistently as a prefix for superseded as well.

**Comment by andrew2net** [comment]
@opoudjis in the source dataset there are the following types of statuses and their possible values:
```ruby
{
  "publicationinfo/pubstatus"=>["Active", "Inactive"],
  "publicationinfo/standard_status"=>["Inactive", "Active", "Superseded"],
  "publicationinfo/holdstatus"=>["Hold", "Publish"],
  "volume/volumeinfo/issue/issuestatus"=>["Complete"],
  "volume/article/articleinfo/articlestatus"=>["Active"],
  "volume/article/articleinfo/holdstatus"=>["Publish"]
}
```
All the found combinations are:
```
Active-Active-Hold-Complete-Active-Publish
Active-Active-Publish-Complete-Active-Publish
Active-Inactive-Hold-Complete-Active-Publish
Active-Inactive-Publish-Complete-Active-Publish
Active-Superseded-Hold-Complete-Active-Publish
Active-Superseded-Publish-Complete-Active-Publish
Inactive-Active-Hold-Complete-Active-Publish
```
It seems all the documents have the same statuses under the `volume` element and we can omit them:
```
Active-Active-Hold
Active-Active-Publish
Active-Inactive-Hold
Active-Inactive-Publish
Active-Superseded-Hold
Active-Superseded-Publish
Inactive-Active-Hold
```
Can we use stages `active | inactive` and substages `active-hold | active-publish | inactive-hold | inactive-publish | superseded-hold | superseded-publish`?

**Comment by ronaldtse** [comment]
@opoudjis FYI the dataset we use is not from screenscraping and we did not make these statuses up. It is the actual [IEEE full bibliographic dataset](https://github.com/relaton/ieee-rawbib), and these statuses are not all seen in the public site.

I don't think we should be attempting to compress these statuses because IEEE SA may change them.

I also want to point out that the statuses mentioned in the OP are of 2 types:
- Project stage and status
- Document stage and status

These two things are technically separate.

Moreover, notice that IEEE SA added statuses for `pubstatus` and `holdstatus`.

We **may** propose a set of "better" categorizations but we must be able to handle what they have now.

**Comment by opoudjis** [comment]
Christ, those statuses are obscene. And you're asking me to make sense of them, only not to.

My suggestion is to break these down so that the primary stage is the standard_status, which seems much more meaningful as we understand standard validity. I don't think the pubstatus and holdstatus are really document maturity stages at all, and they certainly don't act as subcategories of the stage, the way they do in ISO. I propose instead we handle pubstatus and holdstatus as two new ext elements, specific to IEEE: ext/pubstatus and ext/holdstatus. (At most, we could make the pubstatus a substage, but I am inclined just to leave it a separate thing.)

@ronaldtse any disagreement? If not, I'll update the grammar.

**Comment by andrew2net** [comment]
@opoudjis existed values and combinations with `standard_modifier`
```
{
  "publicationinfo/pubstatus"=>["Active", "Inactive"],
  "publicationinfo/standard_status"=>["Inactive", "Active", "Superseded"],
  "publicationinfo/holdstatus"=>["Hold", "Publish"],
  "publicationinfo/standardmodifierset/standard_modifier"=>["Draft", "Withdrawn", "Superseded", "Approved", "Reserved", "Redline"]
}

Active-Active-Hold-Approved
Active-Active-Hold-Draft
Active-Active-Hold-Superseded
Active-Active-Publish
Active-Active-Publish-Approved
Active-Active-Publish-Draft
Active-Active-Publish-Redline
Active-Active-Publish-Superseded
Active-Inactive-Hold-Draft
Active-Inactive-Hold-Withdrawn
Active-Inactive-Publish
Active-Inactive-Publish-Approved
Active-Inactive-Publish-Draft
Active-Inactive-Publish-Redline
Active-Inactive-Publish-Reserved
Active-Inactive-Publish-Withdrawn
Active-Superseded-Hold
Active-Superseded-Publish
Active-Superseded-Publish-Redline
Inactive-Active-Hold-Redline
```
FYI not every document has the `standard_modifier` element.

**Comment by opoudjis** [comment]
Proposal:

stage = standard_status
substage = standard_modified
ext/pubstatus = pubstatus
ext/holdstatus = holdstatus

**Comment by andrew2net** [comment]
> stage = standard_status
> substage = standard_modified
> ext/pubstatus = pubstatus
> ext/holdstatus = holdstatus

@opoudjis @ronaldtse it looks good to me

**Comment by opoudjis** [comment]
Task for me: to update metanorma.org to include these as instructions to authors.

**Comment by opoudjis** [comment]
I will come up with a separate vocabulary of stages for authors, and map these to these statuses.

Draft, Published, Withdrawn, Superseded.

**Comment by ronaldtse** [comment]
The "standardmodifier" values are a random bag of values that mean multiple things.

* Redline: a type of product that is not a "standard"
* Draft: a standard development stage identifier
* Approved, Withdrawn, Superseded: statuses of the standard's applicability
* Reserved: no idea (there are 295 of them)

In Relaton's representation of IEEE metadata, as agreed with @opoudjis , let's keep all 4 categories in `ext/`.

We will have a separate mapping for Relaton values:

* `docstage`: valid values are: `draft`, `approved`, `superseded`, `withdrawn`.
* `doctype`: valid values are: `standard`, `guide`, `whitepaper`, `redline`, `other`. 

Mapping:

IEEE Metadata | Relaton Metadata
--|--
`standardmodifier=Redline` | `doctype=redline`
`standardmodifier=Draft` | `docstage=draft`, `doctype=standard`
`standardmodifier=Approved` | `docstage=approved`, `doctype=standard`
`standardmodifier=Superseded` | `docstage=superseded`, `doctype=standard`
`standardmodifier=Withdrawn` | `docstage=withdrawn`, `doctype=standard`

NOTE: There is only one entry that has `<pubstatus>` of `Inactive` (05948310.xml), so let's ignore it.

**Comment by opoudjis** [comment]
So @andrew2net I am going to add to ext:

standard_status, standard_modified, pubstatus, holdstatus

**Comment by andrew2net** [comment]
fixed in v1.14.6 https://github.com/ietf-tools/relaton-data-ieee/blob/main/data/IEC_IEEE_80005.2.yaml
